base/special/hyperbolic.jl

```diff
@@ -14,6 +14,20 @@
 # is preserved.
 # ====================================================
 
+@inline function exthorner(x, p::Tuple)
+	# polynomial evaluation using compensated summation.
+	# much more accurate, especially when lo can be combined with other rounding errors
+    hi, lo = p[end], zero(x)
+    for i in length(p)-1:-1:1
+        pi = p[i]
+        prod = hi*x
+        err = fma(hi, x, -prod)
```

Comment:
Should we condition use of this on FMA_NATIVE? (Although it's not currently accurate... #33011)

Reply:
I think this should be unconditional. If this is a muladd, llvm will constant fold `err` to 0.  Also are there any relevant platforms without a fast fma? As I understand it, it's required to implement ieee-754 (2008)

Reply:
Makes sense. We should probably discuss to what extent we want to assume fast fma.

Reply:
I've gotten complaints about `fma` causing slow performance before, e.g.: https://github.com/chriselrod/LoopVectorization.jl/issues/134#issuecomment-654192444